File: KITTY/KURS_1/JAVA/DAY_30_1N_ALERTDIALOG/app/src/main/java/com/example/velic/day_30_1_alertdialog/MyDialogFragment4.java
package com.example.velic.day_30_1_alertdialog;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

public class MyDialogFragment4 extends DialogFragment {
    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        final String[] catNamesArray = {"Васька", "Рыжик", "Мурзик"};

        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle("Выберите кота")
                .setItems(catNamesArray, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int which) {
                        Toast.makeText(getActivity(),
                                "Выбранный кот: " + catNamesArray[which],
                                Toast.LENGTH_SHORT).show();
                    }
                });

        return builder.create();
    }
}

File: TEST/DAY_30_1_ALERT_Dialog_kt/app/src/main/java/com/example/day_30_1_alert_dialog_k/MyDialogFragment4.kt
package com.example.day_30_1_alert_dialog_k

import android.app.AlertDialog
import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import android.widget.Toast
import androidx.fragment.app.DialogFragment
//  AlertDialog со списком
class MyDialogFragment4 : DialogFragment() {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {

        val catNamesArray = arrayOf("Васька", "Рыжик", "Мурзик")

        val builder = AlertDialog.Builder(activity)
        builder.setTitle("Выберите кота")
                .setItems(catNamesArray) { dialog, which ->
                    Toast.makeText(activity,
                            "Выбранный кот: " + catNamesArray[which],
                            Toast.LENGTH_SHORT).show()
                }

        return builder.create()
    }
}
